docs/api-reference.md

```diff
@@ -0,0 +1,138 @@
+# Rodux API Reference
+
+## Store
+The Store class is the core piece of Rodux. It is the state container that you create and use.
+
+### Store.new
+```lua
+Store.new(reducer, initialState, middlewares) -> Store
+```
+Creates and returns a new store.
+
+`reducer` is the store's root reducer function. `initialState` is the store's initial state. This should be used to load a saved state from storage. `middlewares` is a table of middlewares to apply.
+
+The store will automatically dispatch an initialization action with a `type` of `@@INIT`.
+
+!!! note
+	The initialization action does not pass through middlewares prior to reaching the reducer.
+
+### Store.changed
+```lua
+Store.changed:connect(function(newState, oldState)
+	-- do something with newState or oldState
+end)
+```
+A [Signal](#Signal) that is fired when the store's state is changed.
+
+!!! danger
+	Do not yield within any listeners on `changed`; an error will be thrown.
+
+### Store:dispatch
+```lua
+Store:dispatch(action) -> nil
+```
+Dispatches an action. The action will travel through all of the store's middlewares before reaching the store's reducer.
+
+The action must contain a `type` field to indicate what type of action it is. No other fields are required.
+
+### Store:getState
+```lua
+Store:getState() -> table
+```
+Gets the store's current state.
+
+!!! warning
+	Do not modify this value. Doing so may cause serious bugs in Rodux, your code, or both!
+
+### Store:destruct
+```lua
+Store:destruct() -> nil
+```
+Destroys the store, disconnecting all connections it may possess.
+
+!!! danger
+	Attempting to use the store after `destruct` has been called will cause problems.
```

Comment:
I wonder if we should add an error for this, like a `_destructed` value on the store.

Reply:
That's an option.

Another is to nuke the store from orbit:

```lua
for key in pairs(self) do
    self[key] = nil
end

setmetatable(self, nil)
```

This has the benefit of making *absolutely sure* that the store isn't hanging onto any leftover values that should be garbage collected. Not sure whether that's worth the possible headache of "I called `destruct` and my store disappeared what?!" that ensues because there's nowhere to throw an error message.